ui-core/src/components/inputs/ComboBox/ComboBox.tsx

```diff
@@ -7,151 +7,139 @@ import React, {
   useMemo,
   useRef,
   useState,
+  useCallback,
 } from 'react';
 
 import { ChevronDown, X } from '@osrd-project/ui-icons';
 import cx from 'classnames';
 
 import { normalizeString } from './utils';
 import Input, { type InputProps } from '../Input';
+import useOutsideClick from '../../../hooks/useOutsideClick';
 
-export type ComboBoxProps<T> = InputProps & {
+export type ComboBoxProps<T> = Omit<InputProps, 'value'> & {
+  value?: T;
   suggestions: Array<T>;
-  getSuggestionLabel: (option: T) => string;
   customLabel?: ReactNode;
   numberOfSuggestionsToShow?: number;
   exactSearch?: boolean;
-  value?: string;
-  onSelectSuggestion?: (option: T | undefined) => void;
   disableDefaultFilter?: boolean;
+  getSuggestionLabel: (option: T) => string;
+  onSelectSuggestion: (option: T | undefined) => void;
+  customOnInputChange?: (inputValue: string) => void;
```

Comment:
This feels a bit awkward when there is already an `onChange` prop that users can set if they need to react to text input changes…

It seems like in OSRD we want to pre-filter suggestions before we pass them to the combobox?

Alternative proposals:

- A `filterSuggestions: boolean` prop to enable/disable the combobox built-in filtering.
- A `filterSuggestions: (suggestions: T[], query: string) => T[]` prop to tweak the combobox built-in filtering. Can be set to `(suggestions, query) => suggestions` to disable filtering.